sor/src/main/java/com/bazaarvoice/emodb/sor/core/DefaultDataStore.java

```diff
@@ -334,12 +339,16 @@ private Resolved resolve(Record record, ReadConsistency consistency) {
     private Expanded expand(final Record record, boolean ignoreRecent, final ReadConsistency consistency) {
         long fullConsistencyTimeStamp = _dataWriterDao.getFullConsistencyTimestamp(record.getKey().getTable());
         long rawConsistencyTimeStamp = _dataWriterDao.getRawConsistencyTimestamp(record.getKey().getTable());
-        return expand(record, fullConsistencyTimeStamp, rawConsistencyTimeStamp, ignoreRecent, consistency);
+        // TODO: right now going with the old stash time just in the current data center - right thing is to get this from all the datacenters.
+        // (Need to address this after confirming if the current data centers approach is the best way to go.)
+        long stashRunTimeStamp = _compactionControlSource.getOldStashTime();
```

Comment:
This is overly broad since it considers the oldest stash time for all placements.  For example, say there is only one Stash going on for placement `ugc_global:ugc` but the record being expanded is in `ugc_eu:ugc`.  There's no reason to restrict deletions on this record since Stash will not read it.

Maybe change the interface to support `_compactionControlSource.getOldestStashTimeForPlacement(record.getKey().getTable().getAvailability().getPlacement())`